bloom-housing/bloom
# feat: new demographics sub-race questions

# Pull Request Template

## Issue Overview

This PR addresses #1739


## Description

Changes demographics to be a series of checkboxes (with sub-categories and occasionally also a custom text input) instead of a dropdown, switches the order of ethnicity and race, and removes gender and sexual orientation from the UI.
Open to changes how the data is being stored, but I just made the race field a string array.
I had to modify the FieldGroup component in some funky ways to be able to support this. In the future we should look into not having every field in a FieldGroup have the same name for every input because that made it hard to target individual inputs to conditionally show subsections and to set default values.

## How Can This Be Tested/Reviewed?

Submit an application from the public side with a set of the new checkboxes. Ensure they exist in partners on the app page, and that you can edit them there. From partners I locally edited the CSV api call in `bloom/sites/partners/pages/listings/[id]/applications/index.tsx` on 136 and added `includeDemographics: true` to test the CSV.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
